tests/testsuite/publish.rs

```diff
@@ -1522,14 +1529,20 @@ Caused by:
 ",
         )
         .run();
-
-    t.join().unwrap();
```

Comment:
If I understand these changes correctly, the servers spawned by the tests are no longer shut down?  I'm concerned that if there are eventually dozens or hundreds of these that might cause issues. Is the fundamental problem not being able to interrupt the server waiting in `accept`?

Reply:
It attempts to shut the server down the `Drop for HttpServerHandle` (`registry.rs` L455), but we can't actually block on the thread joining because we're in `accept`

Reply:
Would it be feasible to open a connection to the server to wake it up so that it can check `self.done` and exit?

Reply:
That seems to work 👍